x-pack/plugin/mapper-constant-keyword/src/main/java/org/elasticsearch/xpack/constantkeyword/mapper/ConstantKeywordFieldMapper.java

```diff
@@ -151,11 +151,14 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
 
         @Override
         public TermsEnum getTerms(boolean caseInsensitive, String string, SearchExecutionContext queryShardContext, String searchAfter) {
+            if (value == null) {
+                return TermsEnum.EMPTY;
+            }
             boolean matches = caseInsensitive
                 ? value.toLowerCase(Locale.ROOT).startsWith(string.toLowerCase(Locale.ROOT))
                 : value.startsWith(string);
             if (matches == false) {
-                return null;
+                return TermsEnum.EMPTY;
```

Comment:
I claim that TermsEnum.EMPTY is the better response here, as null is equivalent to what we return when _terms_enum is not supported on a given field type.

Reply:
PS: I see one of the yaml tests already seems to test "searchAfter", am I right that this curently wouldn't result in an NPE since the value isn't really used on that code path?